packages/components/addon/components/hds/dropdown/list-item/copy-item.js

```diff
@@ -37,12 +38,21 @@ export default class HdsDropdownListItemCopyItemComponent extends Component {
   }
 
   @action
-  copyCode() {
-    navigator.clipboard.writeText(this.args.text);
-    // this if statement resolves to [object Promise] so maybe some improvements
-    // could be made here
-    if (navigator.clipboard.readText()) {
-      this.isSuccess = true;
+  async copyCode() {
+    // If there is text to be copied, then copy it to the clipboard.
+    // If the text in the clipboard is the same as the text to be copied, then set isSuccess to true.
+    // TODO: set any _other_ isSuccess to false, so there is only one indicated successful copy at a time.
+    if (this.args.text) {
+      navigator.clipboard.writeText(this.args.text);
+      navigator.clipboard.readText().then((result) => {
+        if (result === this.args.text) {
+          this.isSuccess = true;
+          // console.log(`msg: ${result} and ${this.args.text} should be equal`);
+        } else {
+          this.isError = true;
+          // console.log(`copyCode was not successful (for whatever reason). One possible reason could be browser clipboard permissions.`);
+        }
+      });
```

Comment:
- since we're using `async/await`, better to use `await` also here
- the rest of the code can be simplified
```suggestion
    const readText = await navigator.clipboard.readText();
    this.isSuccess = readText === this.text;
```

Reply:
Interesting! This code is really hard for me to read, which would make it confusing to maintain. Is there any reason that the code as it's written now is incorrect? Happy to change it if that's the case, but would prefer to keep it a little more readable (and as a result a little easier to maintain) if it's not incorrect. WDYT?